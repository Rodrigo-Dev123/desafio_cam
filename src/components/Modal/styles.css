@import url('https://fonts.googleapis.com/css2?family=Great+Vibes&display=swap');

.btn-modal {
    padding: 10px 20px;
    display: block;
    margin: 5px auto 0;
}

.modal,
.overlay {
    width: 100vw;
    height: 100vh;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    position: fixed;
}

.overlay {
    background: rgba(49, 49, 49, 0.8);
}

.modal-content {
    position: absolute;
    top: 40%;
    left: 50%;
    z-index: 1000;
    transform: translate(-50%, -50%);
    line-height: 1.4;
    background: #f1f1f1;
    padding: 28px 28px;
    border-radius: 10px;
    width: 600px;
    height: 600px;
}

.close-modal {
    border-radius: 0 10px 0  10px ;
    position: absolute;
    top: 0px;
    right: 0px;
    border: none;
    background: red;
    padding: 5px 7px;
    cursor: pointer;
}

.post-info {
    display: flex;
    flex-direction: column;
    gap: 5px;
}

.modal-delete {
    display: block;
    margin: 0 10px;
    border: none;
    background: transparent;
    cursor: pointer;
}

.modal-delete-icon {
    width: 30px;
    height: 30px;
}

.description {
    font-family: 'Great Vibes', cursive;
    font-size: 2rem;
}

.text-modal {
    display: flex;
    margin: 0 auto;
    align-items: center;
    gap: 5;
}

.container-text {
    display: flex;
    flex-direction: column;
}

.modal-delete-icon:hover {
    color: red;
}

@media only screen and (max-width: 600px) {
    
    .modal-content {
        position: absolute;
        top: 40%;
        left: 45%;
        transform: translate(-50%, -50%);
        line-height: 1.4;
        background: #f1f1f1;
        padding: 14px 28px;
        border-radius: 10px;
        width: 300px;
        height: max-content;
        margin: 20px;
    }

    .modal-image {
        width: 100%;
    }
}

@media only screen and (min-width: 600px) {
    .modal-image {
        width: 500px;
        height: 430px;
    }
}