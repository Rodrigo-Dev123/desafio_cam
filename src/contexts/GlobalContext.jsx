import P from 'prop-types';
import { createContext, useState, useEffect } from "react";

export const GlobalContext = createContext();

const GlobalProvider = ({ children }) => {
    const [items, setItems] = useState([]);
    const [itemsPerPage] = useState(10);
    const [currentPage, setCurrentPage] = useState(0);

    const [modal, setModal] = useState(false);
    const [postModal, setPostModal] = useState(null);

    const startIndex = currentPage * itemsPerPage;
    const endIndex = startIndex + itemsPerPage;
    const pages = Math.ceil(items.length / itemsPerPage);
    const currentItems = items.slice(startIndex, endIndex);

    const removeItem = (id) => {
        const updatedItems = JSON.parse(localStorage.getItem('items')).filter(item => {
            return item.id !== id;
        })

        localStorage.setItem('items', JSON.stringify(updatedItems))
        setItems(JSON.parse(localStorage.getItem('items')));
        alert(`Foto ${id} apagada com sucesso! ｡◕‿◕｡`);
    }

    useEffect(() => {
        if (!localStorage.getItem('items')) {
            const fetchData = async () => {
                console.log('fetch execultou!');
                const result = await fetch('https://jsonplaceholder.typicode.com/photos')
                    .then(response => response.json())
                    .then(data => data)

                localStorage.setItem('items', JSON.stringify(result));
                setItems(result);
            }

            fetchData();
            return;
        }
        setItems(JSON.parse(localStorage.getItem('items')));
    }, []);

    return (
        <GlobalContext.Provider value={{
            currentItems, items, setItems, itemsPerPage,
            currentPage, setCurrentPage, modal, setModal,
            startIndex, endIndex, pages, removeItem,
            postModal, setPostModal
        }}>
            {children}
        </GlobalContext.Provider>
    )
}

GlobalProvider.propTypes = {
    children: P.node.isRequired,
};

export default GlobalProvider;
