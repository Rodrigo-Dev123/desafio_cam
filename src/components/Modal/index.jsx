import { useContext } from "react";
import { GlobalContext } from "../../contexts/GlobalContext";
import { AiOutlineClose } from "react-icons/ai";
import { RiDeleteBin6Line } from "react-icons/ri";
import "./styles.css";

export default function Modal() {
  const { modal, removeItem, setModal, postModal } = useContext(GlobalContext);

  const toggleModal = () => {
    setModal(!modal);
  };

  return (
    <>
      {modal && (
        <div className="modal">
          <div onClick={toggleModal} className="overlay"></div>
          <div className="modal-content">
            <div className="">
              {postModal.map((post) => {
                return (
                  <div key={post.id}>
                    <img className="modal-image" src={post.url} alt={post.title}></img>
                    <div className='post-info'>
                      <h1>{post.title}</h1>
                      <p style={ {fontSize: "20px"}}>Foto {post.id}</p>
                      <button className="modal-delete" onClick={() => removeItem(post.id)}>
                        <RiDeleteBin6Line
                          className='modal-delete-icon'
                        />
                      </button>
                    </div>
                  </div>
                )
              })}
            </div>

            <button className="close-modal" onClick={toggleModal}>
              <AiOutlineClose />
            </button>
          </div>
        </div >
      )}
    </>
  );
}