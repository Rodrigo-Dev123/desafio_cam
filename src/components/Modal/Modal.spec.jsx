import { render, screen } from '@testing-library/react';
import { GlobalContext } from '../../contexts/GlobalContext';
import Modal from './index';

const contextValue = {
  removeItem: jest.fn(),
  modal: jest.fn(),
  setModal: jest.fn(),
  postModal: [
    {
      "albumId": 1,
      "id": 1,
      "title": "accusamus beatae ad facilis cum similique qui sunt",
      "url": "https://via.placeholder.com/600/92c952",
      "thumbnailUrl": "https://via.placeholder.com/150/92c952"
    },
  ],
};

describe('<Modal />', () => {
  it('Should render modal', () => {
    const { debug } = render(
      <GlobalContext.Provider value={contextValue}>
        <Modal />
      </GlobalContext.Provider>
    );
    debug();

    expect(screen.getByText(/Foto/i)).toBeInTheDocument();
  });

  it('Shouls math snapshot', () => {
    const { container } = render(
      <GlobalContext.Provider value={contextValue}>
        <Modal />
      </GlobalContext.Provider>
    );
  
    expect(container.firstChild).toMatchSnapshot();
  });
});