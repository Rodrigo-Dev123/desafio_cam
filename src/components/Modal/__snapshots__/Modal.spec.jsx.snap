// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Shouls math snapshot 1`] = `
<div
  class="modal"
>
  <div
    class="overlay"
  />
  <div
    class="modal-content"
  >
    <div
      class=""
    >
      <div>
        <img
          alt="accusamus beatae ad facilis cum similique qui sunt"
          class="modal-image"
          src="https://via.placeholder.com/600/92c952"
        />
        <div
          class="post-info"
        >
          <div
            class="container-text"
          >
            <h1
              class="description"
            >
              accusamus beatae ad facilis cum similique qui sunt
            </h1>
            <div
              class="text-modal"
              style="font-size: 20px;"
            >
              <p>
                Foto 
                1
              </p>
              <button
                class="modal-delete"
              >
                <svg
                  class="modal-delete-icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 4V2H17V4H22V6H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V6H2V4H7ZM6 6V20H18V6H6ZM9 9H11V17H9V9ZM13 9H15V17H13V9Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="close-modal"
    >
      <svg
        fill="currentColor"
        fill-rule="evenodd"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z"
        />
      </svg>
    </button>
  </div>
</div>
`;
